pkg/cli/create_helm.go

```diff
@@ -215,17 +240,23 @@ func CreateHelm(ctx context.Context, options *CreateOptions, globalFlags *flags.
 		cfg := &config.Config{}
 		err = cfg.UnmarshalYAMLStrict(data)
 		if err != nil {
-			if errors.Is(err, config.ErrInvalidConfig) {
-				cmd.log.Infof("If you are using the old values format, consider using %q to convert it to the new v0.20 format", "vcluster convert config")
+			if !errors.Is(err, config.ErrInvalidConfig) {
+				return err
 			}
-			return err
+			// TODO Delete after vCluster 0.19.x resp. the old config format is out of support.
+			// We cannot discriminate between k0s/k3s and eks/k8s. So we cannot prompt the actual values to convert, as this would cause false positives,
+			// because users are free to e.g. pass a k0s values file to a currently running k3s virtual cluster.
+			if isLegacyConfig(data) {
+				return fmt.Errorf("it appears you are using a vCluster configuration using pre-v0.20 formatting. Please run %q to convert the values to the latest format", "vcluster convert config")
+			}
+			// TODO end
 		}
```

Comment:
I would maybe structure the if differently here 
```go
if errors.Is(err, config.ErrInvalidConfig) {
  if isLegacyConfig(data){
    return fmt.Errorf("it appears you are using a vCluster configuration using pre-v0.20 formatting. Please run %q to convert the values to the latest format", "vcluster convert config")
  }
} else if err != nil {
  return err
}
```
that is simpler to follow imho

Reply:
**Update**: The `ErrInvalidConfig` did not provide any benefit. It was actually a left-over from when the config unmarshaling looked different. So I changed it to a simple err check.